wrangler-service/src/main/java/co/cask/wrangler/service/gcs/GCSService.java

```diff
@@ -510,12 +511,12 @@ public void specification(HttpServiceRequest request, final HttpServiceResponder
     return whitelist;
   }
 
-  private Collection<Bucket> getBuckets(Storage storage, Set<String> whitelist) {
+  private Collection<Bucket> getBuckets(Storage storage, Set<String> whitelist, int bucketLimit) {
     // this will include buckets that can be listed by the service account, but may not include all buckets
     // in the whitelist, if the whitelist contains publicly accessible buckets from other projects.
     // do some post-processing to filter out anything not in the whitelist and also try and lookup buckets
     // that are in the whitelist but not in the returned list
-    Page<Bucket> list = storage.list();
+    Page<Bucket> list = storage.list(Storage.BucketListOption.pageSize(bucketLimit));
```

Comment:
this sounds like it's a page size limit, not a limit for the entire response. list.iterateAll() will iterate through every page. There is another method that iterates through just the elements in the page.

Is there a limit for how big the requested page size can be? If not, we just need to change the iterateAll() to the other one. If there is, we need to go page by page and limit it ourselves.